src/main/java/io/anserini/fusion/ScoredDocsFuser.java

```diff
@@ -0,0 +1,270 @@
+        scoredDocs.scores[i] = score;
+    }
+  }
+
+  private static void rescoreScale(double scale, ScoredDocs scoredDocs) {
+    int length = scoredDocs.lucene_documents.length;
+    for(int i = 0; i < length; i++){
+        float score = (float) (scoredDocs.scores[i] * scale);
+        scoredDocs.scores[i] = score;
+    }
+  }
+
+  private static void normalizeScores(ScoredDocs scoredDocs) {
+    Map<String, List<Integer>> indicesForTopics = new HashMap<String, List<Integer>>(); // topic, list of indices for that topic
+    int length = scoredDocs.lucene_documents.length;
+    for(int i = 0; i < length; i++){
+        indicesForTopics.computeIfAbsent(scoredDocs.lucene_documents[i].get(TOPIC), k -> new ArrayList<>()).add(i);
+    }
+
+    for(List<Integer> topicIndices : indicesForTopics.values()){
+        int numRecords = topicIndices.size();
+        float minScore = scoredDocs.scores[topicIndices.get(0)];
+        float maxScore = scoredDocs.scores[topicIndices.get(numRecords - 1)];
+        for(int i = 0; i < numRecords; i++){
+            int index = topicIndices.get(i);
+            minScore = Float.min(minScore, scoredDocs.scores[index]);
+            maxScore = Float.max(maxScore, scoredDocs.scores[index]);
+        }
+
+        for(int i = 0; i < numRecords; i++){
+            int index = topicIndices.get(i);
+            float normalizedScore = ((float) scoredDocs.scores[index] - minScore) / (maxScore - minScore);
+            scoredDocs.scores[index] = normalizedScore;
+        }
+    }
+  }
+
+  /**
+   * Merges multiple ScoredDocs instances into a single ScoredDocs instance.
+   * The merged ScoredDocs will contain the top documents for each topic, with scores summed across the input runs.
+   *
+   * @param runs  List of ScoredDocs instances to merge.
+   * @param depth Maximum number of documents to consider from each run for each topic (null for no limit).
+   * @param k     Maximum number of top documents to include in the merged run for each topic (null for no limit).
+   * @return A new ScoredDocs instance containing the merged results.
+   * @throws IllegalArgumentException if less than 2 runs are provided.
+   */
+  public static ScoredDocs merge(List<ScoredDocs> runs, Integer depth, Integer k) {
+    if (runs.size() < 2) {
+      throw new IllegalArgumentException("Merge requires at least 2 runs.");
+    }
+
+    // for every topic, produce a map of docid to score, num of accumulated
+    HashMap<String, HashMap<String, AbstractMap.SimpleEntry<Float, Integer>>> docScores = new HashMap<>();
+    for(ScoredDocs run : runs){
+      for(int i = 0; i < run.lucene_documents.length; i++){
+        String query = run.lucene_documents[i].get(TOPIC);
+        String docid = run.docids[i];
+        Float score = run.scores[i];
+        docScores.computeIfAbsent(query, key -> new HashMap<>())
+          .merge(docid, new AbstractMap.SimpleEntry<>(score, 1), 
```

Comment:
I suspect it's parts like this that are slow... manipulating `HashMap`s creates lots of objects.

Nothing actionable, just noting for a future iteration.